modules/axruntime/src/mp.rs

```diff
@@ -1,28 +1,39 @@
 use core::sync::atomic::{AtomicUsize, Ordering};
 
-use axconfig::{SMP, TASK_STACK_SIZE};
+use axconfig::{CPU_ID_LIST, SMP, TASK_STACK_SIZE};
 use axhal::mem::{virt_to_phys, VirtAddr};
 
 #[link_section = ".bss.stack"]
 static mut SECONDARY_BOOT_STACK: [[u8; TASK_STACK_SIZE]; SMP - 1] = [[0; TASK_STACK_SIZE]; SMP - 1];
 
 static ENTERED_CPUS: AtomicUsize = AtomicUsize::new(1);
 
+#[allow(clippy::absurd_extreme_comparisons)]
+fn start_secondary_cpus_one(primary_cpu_id: usize, cpu_id: usize, cpu_i: &mut usize) {
+    if cpu_id != primary_cpu_id && *cpu_i < SMP - 1 {
+        let stack_top = virt_to_phys(VirtAddr::from(unsafe {
+            SECONDARY_BOOT_STACK[*cpu_i].as_ptr_range().end as usize
+        }));
+
+        debug!("starting CPU {}...", cpu_id);
+        axhal::mp::start_secondary_cpu(cpu_id, stack_top);
+        *cpu_i += 1;
+
+        while ENTERED_CPUS.load(Ordering::Acquire) <= *cpu_i {
+            core::hint::spin_loop();
+        }
+    }
+}
+
 pub fn start_secondary_cpus(primary_cpu_id: usize) {
-    let mut logic_cpu_id = 0;
-    for i in 0..SMP {
-        if i != primary_cpu_id {
-            let stack_top = virt_to_phys(VirtAddr::from(unsafe {
-                SECONDARY_BOOT_STACK[logic_cpu_id].as_ptr_range().end as usize
-            }));
-
-            debug!("starting CPU {}...", i);
-            axhal::mp::start_secondary_cpu(i, stack_top);
-            logic_cpu_id += 1;
-
-            while ENTERED_CPUS.load(Ordering::Acquire) <= logic_cpu_id {
-                core::hint::spin_loop();
-            }
+    let mut cpu_i = 0;
+    if CPU_ID_LIST.is_empty() {
```

Comment:
Move these logic into `axhal/src/platform/xxx/mp.rs`, the `cpu_id` is designed as a logical CPU ID (`0`..`SMP`) in this context.